parser/structured_data.go

```diff
@@ -51,53 +51,46 @@ func (sd *StructuredData) Parse(bytes []byte) (*skogul.Container, error) {
 	}, nil
 }
 
-// mnrParseData takes the raw input and parses it
+// parseStructuredData takes the raw input and parses it
 // this takes care of splitting input on newlines etc
 func (sd *StructuredData) parseStructuredData(data []byte) ([]*skogul.Metric, error) {
-	lines := bytes.Split(data, []byte("\n"))
+	lineScanner := bufio.NewScanner(bytes.NewReader(data))
```

Comment:
Ah, but it'ls not technically splitting on lines any more?

Reply:
Well, we kind of are: https://github.com/telenornms/skogul/blob/d311a6bb03335f4704ec0e250ef0bbf7653b79a2/parser/structured_data.go#L174-L178

We just don't split the data on newlines *first*, and *then* start looking for closing `]`'s.

We loop until we find an unescaped `]` or `\n`, it doesn't matter which. And if we get a `]\n` we'll just short-circuit the next iteration and start parsing from the next char if there are more left.